Update notebooks

jserranohidalgo opened the pull request:
Notebook review

jserranohidalgo commented:
@alfonsorr the notebook for joins is still pending

jeslg commented:
Some (minor) comments about README notebook:
* `Case 2: Avoid implicit type castings`: I think that the term "implicit" could lead to confusion. I'd say something like `Avoid unsolicited type castings`.
* `Sacrifying`: sacrificing?
* `with respect conventional`: with respect to conventional.
* `def df = (1 to 1000).toDF.select($"value"+f.lit(true))` Is it necessary to replace `List(1, 2, 3)` by `(1 to 1000)`?
* `col[Int]("value") * lit(true)` use `+` instead to stick to the original example
* `List(1,2,3).toDF.withColumn("other", colInt("value") * lit(1))`: you started using `col[Int]` and now moved to `colInt`. I think it's preferible to stick to the same convention here.
* I'd use `foo` instead of `jander` 😄 
* The import `doricf` becomes noisy, could we use a wildcard import instead? You have already renamed the std one to `f`... Besides, prefixing some instructions with `doricf` (as in `doricf.concat`) but keeping others with their simple name (as in `colInt`) is weird.
* `it's of an unexpected type`: it sounds too Spanish... maybe it's correct!

I like it! The usage cases motivate really well the need of the library.

jserranohidalgo commented:
@jeslg The expression "implicit type cast" is not uncommon in Spark: e.g.,  https://issues.apache.org/jira/browse/SPARK-18174 and
![image](https://user-images.githubusercontent.com/3715323/122600082-91bd9480-d06f-11eb-923b-3bb82b258b2a.png)
in the "definitive guide" book

jserranohidalgo commented:
> * `Sacrifying`: sacrificing?
> * `with respect conventional`: with respect to conventional.
> * `def df = (1 to 1000).toDF.select($"value"+f.lit(true))` Is it necessary to replace `List(1, 2, 3)` by `(1 to 1000)`?
> * `col[Int]("value") * lit(true)` use `+` instead to stick to the original example

👍 

> * `List(1,2,3).toDF.withColumn("other", colInt("value") * lit(1))`: you started using `col[Int]` and now moved to `colInt`. I think it's preferible to stick to the same convention here.

I start with the former because it's more foundamental, and change then to the latter because it's less noisy for most people ... perhaps. In any case, I added the comment "We will use these aliases in the sequel."

> * The import `doricf` becomes noisy, could we use a wildcard import instead? You have already renamed the std one to `f`... Besides, prefixing some instructions with `doricf` (as in `doricf.concat`) but keeping others with their simple name (as in `colInt`) is weird.

Yes, you're right. In the new refactoring this won't happen

> * `it's of an unexpected type`: it sounds too Spanish... maybe it's correct!

fixed

> 
> I like it! The usage cases motivate really well the need of the library.

credits to @alfonsorr :)

jserranohidalgo commented:
@alfonsorr please, review and merge if you think it's right

alfonsorr commented on README.md:
```diff
@@ -1,71 +1,54 @@
+
+[![CI](https://github.com/hablapps/doric/actions/workflows/ci.yml/badge.svg)](https://github.com/hablapps/doric/actions/workflows/ci.yml)
+[![Scala Steward badge](https://img.shields.io/badge/Scala_Steward-helping-blue.svg?style=flat&logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAQCAMAAAARSr4IAAAAVFBMVEUAAACHjojlOy5NWlrKzcYRKjGFjIbp293YycuLa3pYY2LSqql4f3pCUFTgSjNodYRmcXUsPD/NTTbjRS+2jomhgnzNc223cGvZS0HaSD0XLjbaSjElhIr+AAAAAXRSTlMAQObYZgAAAHlJREFUCNdNyosOwyAIhWHAQS1Vt7a77/3fcxxdmv0xwmckutAR1nkm4ggbyEcg/wWmlGLDAA3oL50xi6fk5 ...
+[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/hablapps/doric/HEAD?filepath=notebooks/README.ipynb)
```
Change the link to binder to not point directly to a notebook, the joins notebook will feel very lonely

alfonsorr commented:
Love it!!! 😍
